### PR TITLE
[14528] Add documentation to the Fast DDS-Gen builtin annotations

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Apart from adding documentation to the builtin annotations, this PR also fixes annotations `@value` and `@default_literal` which are still unimplemented.

Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

### CODE REVIEW

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 1]
```diff
@@ -402,7 +402,7 @@ Additionally, the following standard annotations are builtin (recognized and pas
 +-------------------------+-------------------------------------------------------------------------+
 | @position               | Used by bitmasks_.                                                      |
 +-------------------------+-------------------------------------------------------------------------+
-| @value                  | Allows to set a constant value to any element.                          |
+| @value                  | Unimplemented.                                                          |
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:405]: I wouldn't remove the description
```suggestion
| @value                  | [Unimplemented] Allows to set a constant value to any element.   |
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
I think this ought to be changed in Fast DDS-Gen help as well

**Comment by JLBuenoLopez-eProsima** [comment]
Which help are you referring to? Fast DDS-Gen help only refers to the tool options and not to the annotation support or other implementation details. I do not think we should change this.

**Review by EduPonz — APPROVED** [review]
LGTM

**Pull request merged**